src/auth/index.js

```diff
@@ -32,6 +32,9 @@ export default class Auth {
     this.platform = options.platform || _platform;
     this.storage = createStorage(options);
     this.providers = createProviders(this, options);
+    this._willInitialize = this._get().then(auth => {
```

Comment:
It seems like you may want a similar pattern for a static method on Auth to initialize an instance of it with no constructor such that we can get rid of these initialization variables